tools/rosbag/src/rosbag/bag.py

```diff
@@ -857,7 +857,7 @@ def get_type_and_topic_info(self, topic_filters=None):
         # load our list of topics and optionally filter
         if topic_filters is not None:
             if not isinstance(topic_filters, list):
-                topic_filters = [topic_filters]
+                topic_filters = list(topic_filters)
```

Comment:
The doc string says `topic_filters` may either be a string, or a list of strings.

```
@type topic_filters: either a single str or a list of str.
```

If given `topic_filters = "foo"`, the line `topic_filters = [topic_filters]` makes `topic_filters` =  `["foo"]`, but IIUC this change would make it equal to `["f", "o", "o"]`.

Reply:
Ok, but it already 2 years for this PR. Please make fixes anyway you like 👍

Reply:
Thank you for your patience these two years. Unfortunately I don't have the bandwidth to make and test the changes, so I'll close this for now.